src_c/transform.c

```diff
@@ -2966,6 +2966,152 @@ surf_average_color(PyObject *self, PyObject *args, PyObject *kwargs)
     return Py_BuildValue("(bbbb)", r, g, b, a);
 }
 
+static void
+box_blur(SDL_Surface *src, SDL_Surface *dst, int radius)
```

Comment:
It takes about 100ms to process a image using this algorithm.

Image: 2048x2048 RGBA
CPU: AMD Ryzen7 6800H